<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RuleConfigDialog</class>
 <widget class="QDialog" name="RuleConfigDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1112</width>
    <height>669</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DAS Monitor Rule Configuration</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3" stretch="1,0">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Rules:</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QToolButton" name="addRuleButton">
           <property name="font">
            <font>
             <weight>50</weight>
             <bold>false</bold>
             <kerning>true</kerning>
            </font>
           </property>
           <property name="text">
            <string>Add Rule</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="removeRuleButton">
           <property name="font">
            <font>
             <weight>50</weight>
             <bold>false</bold>
             <kerning>true</kerning>
            </font>
           </property>
           <property name="text">
            <string>Delete Rule</string>
           </property>
           <property name="popupMode">
            <enum>QToolButton::InstantPopup</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTableWidget" name="ruleTable">
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="cornerButtonEnabled">
          <bool>false</bool>
         </property>
         <property name="columnCount">
          <number>3</number>
         </property>
         <attribute name="horizontalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <column/>
         <column/>
         <column/>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Signals:</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QToolButton" name="addSignalButton">
           <property name="font">
            <font>
             <weight>50</weight>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>Add Signal</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="removeSignalButton">
           <property name="font">
            <font>
             <weight>50</weight>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>Delete Signal</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTableWidget" name="signalTable">
         <property name="alternatingRowColors">
          <bool>true</bool>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="cornerButtonEnabled">
          <bool>false</bool>
         </property>
         <property name="columnCount">
          <number>6</number>
         </property>
         <attribute name="horizontalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderStretchLastSection">
          <bool>false</bool>
         </attribute>
         <column/>
         <column/>
         <column/>
         <column/>
         <column/>
         <column/>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="layoutWidget_2">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Facts:</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_9">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QToolButton" name="refreshFactButton">
           <property name="text">
            <string>Refresh</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="factFilterCB"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QListWidget" name="factList">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="statusLabel">
       <property name="minimumSize">
        <size>
         <width>120</width>
         <height>0</height>
        </size>
       </property>
       <property name="font">
        <font>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="frameShape">
        <enum>QFrame::Box</enum>
       </property>
       <property name="text">
        <string>Status</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QToolButton" name="getCfgButton">
       <property name="text">
        <string>Get</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="setCfgButton">
       <property name="text">
        <string>Set</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_4">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QToolButton" name="restoreDefCfgButton">
       <property name="text">
        <string>Restore Default</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="setDefCfgButton">
       <property name="text">
        <string>Set Default</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="helpButton">
       <property name="text">
        <string>Help</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="okButton">
       <property name="text">
        <string>Ok</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>okButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>879</x>
     <y>441</y>
    </hint>
    <hint type="destinationlabel">
     <x>465</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cancelButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>788</x>
     <y>441</y>
    </hint>
    <hint type="destinationlabel">
     <x>465</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>helpButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>showHelp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>715</x>
     <y>672</y>
    </hint>
    <hint type="destinationlabel">
     <x>540</x>
     <y>347</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>refreshFactButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>getFacts()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>900</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>471</x>
     <y>346</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>getCfgButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>getRules()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>46</x>
     <y>599</y>
    </hint>
    <hint type="destinationlabel">
     <x>555</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setCfgButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>setRules()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>88</x>
     <y>599</y>
    </hint>
    <hint type="destinationlabel">
     <x>555</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>restoreDefCfgButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>getDefaultRules()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>396</x>
     <y>599</y>
    </hint>
    <hint type="destinationlabel">
     <x>555</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setDefCfgButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>setDefaultRules()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>499</x>
     <y>599</y>
    </hint>
    <hint type="destinationlabel">
     <x>555</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RuleConfigDialog</sender>
   <signal>writeAllowed(bool)</signal>
   <receiver>setCfgButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>88</x>
     <y>599</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RuleConfigDialog</sender>
   <signal>readAllowed(bool)</signal>
   <receiver>getCfgButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>46</x>
     <y>599</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RuleConfigDialog</sender>
   <signal>readAllowed(bool)</signal>
   <receiver>setDefCfgButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>599</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RuleConfigDialog</sender>
   <signal>readAllowed(bool)</signal>
   <receiver>restoreDefCfgButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>599</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RuleConfigDialog</sender>
   <signal>readAllowed(bool)</signal>
   <receiver>removeSignalButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>472</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addRuleButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>addRule()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>905</x>
     <y>106</y>
    </hint>
    <hint type="destinationlabel">
     <x>465</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addSignalButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>addSignal()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>905</x>
     <y>316</y>
    </hint>
    <hint type="destinationlabel">
     <x>465</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeRuleButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>removeSelectedRule()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>905</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>465</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RuleConfigDialog</sender>
   <signal>readAllowed(bool)</signal>
   <receiver>removeRuleButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>197</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeSignalButton</sender>
   <signal>clicked()</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>removeSelectedSignal()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>905</x>
     <y>346</y>
    </hint>
    <hint type="destinationlabel">
     <x>465</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RuleConfigDialog</sender>
   <signal>readAllowed(bool)</signal>
   <receiver>addRuleButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RuleConfigDialog</sender>
   <signal>readAllowed(bool)</signal>
   <receiver>addSignalButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>442</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RuleConfigDialog</sender>
   <signal>readAllowed(bool)</signal>
   <receiver>signalTable</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>271</x>
     <y>456</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>RuleConfigDialog</sender>
   <signal>readAllowed(bool)</signal>
   <receiver>ruleTable</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>271</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>factFilterCB</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>RuleConfigDialog</receiver>
   <slot>setFactFilter(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1058</x>
     <y>24</y>
    </hint>
    <hint type="destinationlabel">
     <x>555</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>writeAllowed(bool)</signal>
  <signal>readAllowed(bool)</signal>
  <slot>addRule()</slot>
  <slot>removeSelectedRule()</slot>
  <slot>addSignal()</slot>
  <slot>removeSelectedSignal()</slot>
  <slot>getRules()</slot>
  <slot>setRules()</slot>
  <slot>showHelp()</slot>
  <slot>getFacts()</slot>
  <slot>getDefaultRules()</slot>
  <slot>setDefaultRules()</slot>
  <slot>setFactFilter(int)</slot>
 </slots>
</ui>
